ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/part/projectexplorer/ProjectExplorerPartPresenter.java

```diff
@@ -340,10 +340,13 @@ public void onContextMenu(int mouseX, int mouseY) {
     /** {@inheritDoc} */
     @Override
     public void onDeleteKey() {
-        final TreeNode<?> selectedNode = view.getSelectedNode();
-        if (selectedNode instanceof StorableNode) {
-            deleteNodeHandler.delete((StorableNode)selectedNode);
+        List<StorableNode> selectedNodes = new ArrayList<>();
```

Comment:
I think it would be better to use Guava library for filtration list of nodes, smth like this:

```java
        Array<TreeNode<?>> selectedNodes = view.getSelectedNode();

        List<StorableNode> storableNodes = FluentIterable.from(selectedNodes.asIterable())
                .filter(new Predicate<TreeNode<?>>() {
                    @Override
                    public boolean apply(TreeNode<?> node) {
                        return node instanceof StorableNode;
                    }
                }).transform(new Function<TreeNode<?>, StorableNode>() {
                    @Nullable
                    @Override
                    public StorableNode apply(@Nullable TreeNode<?> node) {
                        return (StorableNode)node;
                    }
                }).copyInto(new ArrayList<StorableNode>());

                deleteNodeHandler.deleteNodes(storableNodes);
```

@vparfonov WDYT?